pipeline/metadata/schema.py

```diff
@@ -0,0 +1,299 @@
+  matches_control: Optional[str] = None
+
+  def add_tag_to_answer(self, tag: SatelliteAnswerMetadata) -> None:
+    """Add tag information to a Satellite answer"""
+    if tag.asnum is not None:
+      self.asnum = tag.asnum
+    if tag.asname is not None:
+      self.asname = tag.asname
+    if tag.http is not None:
+      self.http = tag.http
+    if tag.cert is not None:
+      self.cert = tag.cert
+    if tag.matches_control is not None:
+      self.matches_control = tag.matches_control
+
+
+@dataclass
+class SatelliteAnswerMetadata(SatelliteAnswer):
+  """Satellite Answer Metadata.
+
+  When this metadata is being passed around
+  it needs an additional date field to keep track of when it's valid.
+  In the final SatelliteAnswer object written to bigquery
+  we don't include that field since it's redundant.
+  """
+  date: str = ''
+
+
+@dataclass
+class IpMetadata():
+  """Class for keeping track of ip metadata"""
+  # Keys
+  ip: str
+  date: str
+  # Metadata
+  netblock: Optional[str] = None
+  asn: Optional[int] = None
+  as_name: Optional[str] = None
+  as_full_name: Optional[str] = None
+  as_class: Optional[str] = None
+  country: Optional[str] = None
+  organization: Optional[str] = None
+  # Satellite Metadata
+  name: Optional[str] = None
+
+
+@dataclass
+class ReceivedHttps:
+  """Class for the parsed content of a received HTTP/S request
+
+  These are both passed around independantly, and as part of
+  HyperquackRow/BlockpageRow objects
+  """
+  blockpage: Optional[bool] = None
+  page_signature: Optional[str] = None
+
+  received_status: Optional[str] = None
+  received_body: Optional[str] = None
+  received_tls_version: Optional[int] = None
+  received_tls_cipher_suite: Optional[int] = None
+  received_tls_cert: Optional[str] = None
+  received_headers: List[str] = dataclasses.field(default_factory=list)
+
+  def update_received(self, new: ReceivedHttps) -> None:
+    """Imitation update method that matches the semantics of python's dict update
+
+    Both HyperquackRow and BlockpageRow use this method
+    to add in ReceivedHttps fields to themselves.
+    """
+    if new.blockpage is not None:
+      self.blockpage = new.blockpage
+    if new.page_signature is not None:
+      self.page_signature = new.page_signature
+    if new.received_status is not None:
+      self.received_status = new.received_status
+    if new.received_body is not None:
+      self.received_body = new.received_body
+    if new.received_tls_version is not None:
+      self.received_tls_version = new.received_tls_version
+    if new.received_tls_cipher_suite is not None:
+      self.received_tls_cipher_suite = new.received_tls_cipher_suite
+    if new.received_tls_cert is not None:
+      self.received_tls_cert = new.received_tls_cert
+    if new.received_headers != []:
+      self.received_headers = new.received_headers
+
+
+# All or part of a scan row to be written to bigquery
+@dataclass
+class Row:  # Corrosponds to BASE_BIGQUERY_SCHEMA
```

Comment:
Maybe call it `BigqueryRow` to make it clearer this is the output row (not the input).

Reply:
done